### PR TITLE
Add twitch badges

### PR DESCRIPTION
**Pull request by balroggg**
Add support for twitch badges by adding some emotes in the beginning of
the username.

Support badges:
- vip :gem: 
- moderator :hammer: 
- subscriber :star: 
- prime gaming :crown:

### CODE REVIEW

**Comment by Xithrius** [comment]
Testing this out, works very well.

Next step would most likely be a CLI argument and a config value being available to disable/enable badges.

File: src/twitch.rs [hunk 1]
```diff
@@ -105,15 +105,57 @@ pub async fn twitch_irc(mut config: CompleteConfig, tx: Sender<Data>, mut rx: Re
                                 Some(username) => username.to_string(),
                                 None => "Undefined username".to_string(),
                             };
-                            // try to get username from message tags
+                            let mut badges = String::new();
                             if let Some(ref tags) = message.tags {
+                                const VIP_BADGE: char = '\u{1F48E}';
+                                const MODERATOR_BADGE: char = '\u{1F528}';
+                                const SUBSCRIBER_BADGE: char = '\u{2B50}';
+                                const PRIME_GAMING_BADGE: char = '\u{1F451}';
```
Comment (Xithrius) [inline src/twitch.rs:113]: Constants should be moved to top of file

File: src/utils/text.rs [hunk 2]
```diff
@@ -9,18 +10,18 @@ pub fn align_text(text: &str, alignment: &str, maximum_length: u16) -> String {
...
-            "{text:^width$}",
-            text = text,
-            width = std::cmp::max(maximum_length, text.len() as u16) as usize
-        ),
+        "right" => format!("{}{}", " ".repeat(maximum_length as usize - dw), text),
+        "center" => {
+            let side_spaces =
+                " ".repeat(((maximum_length / 2) - (((dw / 2) as f32).floor() as u16)) as usize);
+            format!("{}{}{}", side_spaces, text, side_spaces)
+        }
```
Comment (Xithrius) [inline src/utils/text.rs:24]: I was worried that this wasn't implemented before reviewing, thanks for realizing that this needed changes.

**Review by Xithrius — CHANGES_REQUESTED** [review]
A singular change.

File: src/twitch.rs (hunk 1 above)
Comment (balroggg) [reply src/twitch.rs]: Ok

File: src/utils/text.rs (hunk 2 above)
Comment (balroggg) [reply src/utils/text.rs:24]: I wanted a general solution, but this doesn't work with emotes(

Comment (Xithrius) [reply src/utils/text.rs:24]: Unfortunate.

**Comment by Xithrius** [comment]
Looping back to my comment before the review, would you like to implement the config values for enabling/disabling badges?

**Comment by balroggg** [comment]
Ok,  I will do it, to not split reviews.

**Comment by Xithrius** [comment]
Very well.

**Comment by balroggg** [comment]
Add badges cli arg
```
# badges on
cargo run -- -c ixxdeee -b
# badges off
cargo run -- -c ixxdeee
```
Enabled in default-config.toml by default

File: src/handlers/args.rs [hunk 3]
```diff
@@ -33,6 +33,10 @@ pub struct Cli {
     /// Username color palette
     #[clap(short, long, possible_values = &["pastel", "vibrant", "warm", "cool"])]
     pub palette: Option<Palette>,
+
+    /// Twitch badges support
+    #[clap(short, long)]
+    pub badges: bool,
```
Comment (Xithrius) [inline src/handlers/args.rs:39]:
```suggestion
    pub badges: Option<bool>,
```

File: src/handlers/args.rs [hunk 4]
```diff
@@ -62,4 +66,5 @@ pub fn merge_args_into_config(config: &mut CompleteConfig, args: Cli) {
     if let Some(palette) = args.palette {
         config.frontend.palette = palette;
     }
+    config.frontend.badges = args.badges;
```
Comment (Xithrius) [inline src/handlers/args.rs:69]:
```suggestion
    if let Some(badges) = args.badges {
        config.frontend.badges = badges;
    }
```

**Review by Xithrius — CHANGES_REQUESTED** [review]
So the CLI doesn't explicitly have to pass in a value.

**Review by Xithrius — APPROVED** [review]

**Comment by Xithrius** [comment]
Thanks for this PR!

**Pull request merged**